### PR TITLE
fix(blocks): unexpected newline appears while pasting multiply-paragr…

### PR DESCRIPTION
**Pull request by raintoway**
…aph-blocks to code-block

[issue](https://github.com/toeverything/blocksuite/issues/8150)

effect as follow

### CODE REVIEW

File: packages/blocks/src/_common/adapters/plain-text.ts [hunk 1]
```diff
@@ -232,6 +232,7 @@ export class PlainTextAdapter extends BaseAdapter<PlainText> {
       return null;
     }
     payload.file = payload.file.replaceAll('\r', '');
+    payload.file = payload.file.replace(/\n$/, '');
```
Comment (L-Sun) [inline packages/blocks/src/_common/adapters/plain-text.ts:235]: I think the reason is in the exporter side, not here. @fourdim PTAL.

Comment (raintoway) [reply packages/blocks/src/_common/adapters/plain-text.ts:235]: yeah, i think so too ,  here is the export side in `/packages/blocks/src/_common/adapters/plain-text.ts`
```ts
  private async _traverseSnapshot(
    snapshot: BlockSnapshot
  ): Promise<{ mixtext: string }> {
    ...
    walker.setEnter(o => {
      const text = (o.node.props.text ?? { delta: [] }) as {
        delta: DeltaInsert[];
      };
      switch (o.node.flavour) {
        ...
        case 'affine:paragraph': {
          buffer += text.delta.map(delta => delta.insert).join('');
          buffer += '\n';
          break;
        }
 ...
```
but i think change here will affect a lot , so i only change the adapter

Comment (fourdim) [reply packages/blocks/src/_common/adapters/plain-text.ts:235]: No, changing the adapter will affect a lot.

> It's not about adding an extra newline at the end of a file, it's about not removing the newline that should be there.

> The C standard specifies that a C file should end with a newline (C11, 5.1.1.2, 2.) and that a last line without a newline yields undefined behavior (C11, J.2, 2nd item).

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

**Review by fourdim — CHANGES_REQUESTED** [review]

File: packages/blocks/src/_common/adapters/plain-text.ts (hunk 1 above)
Comment (fourdim) [reply packages/blocks/src/_common/adapters/plain-text.ts:235]: No, changing this adapter will affect a lot.

> It's not about adding an extra newline at the end of a file, it's about not removing the newline that should be there.

> The C standard specifies that a C file should end with a newline (C11, 5.1.1.2, 2.) and that a last line without a newline yields undefined behavior (C11, J.2, 2nd item).

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

**Review by fourdim — CHANGES_REQUESTED** [review]

**Comment by fourdim** [comment]
Closed by #8575.

**Pull request closed without merging**